nifi-extension-bundles/nifi-snmp-bundle/nifi-snmp-processors/src/main/java/org/apache/nifi/snmp/operations/SNMPTrapReceiverHandler.java

```diff
@@ -69,7 +71,12 @@ public int getListeningPort() {
     }
 
     public void createTrapReceiver(final ProcessSessionFactory processSessionFactory, final ComponentLog logger) {
-        addUsmUsers();
+        if (configuration.getVersion() == SnmpConstants.version3) {
+            removeMessageProcessingModel(MessageProcessingModel.MPv1);
+            removeMessageProcessingModel(MessageProcessingModel.MPv2c);
```

Comment:
@Lehel44 Thanks for this improvement! So unauthenticated v1 and v2 traps are now disabled. However, it is still possible to send v3 traps with a `noAuthNoPriv` client, even if `ListTrapSNMP` would require `authPriv`. Could you please check this case too?